### PR TITLE
Updated cucumber yml and tests

### PR DESCRIPTION
**Pull request by karthikmsqa**
# Description

This PR includes the following proposed change(s):

- Updated docker compose with frontend changes
- Updated cucumber workflow yml

## Type of change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

On docker demo mode locally and on push to the branch. 

## Checklist:

- [x] I have performed a self-review of my own code

### CODE REVIEW

**Review by sdevalapurkar-bcgov — DISMISSED** [review]

File: .github/workflows/cucumber-tests.yml [hunk 1]
```diff
@@ -19,17 +19,27 @@ jobs:
...
-        run: KEYCLOAK_CREDENTIALS_SECRET=${{secrets.KEYCLOAK_CREDENTIALS_SECRET}} KEYCLOAK_REALM=${{secrets.KEYCLOAK_REALM}} KEYCLOAK_URL=${{secrets.KEYCLOAK_URL}} EFILING_DEMO_KEYCLOAK_REALM=${{secrets.EFILING_DEMO_KEYCLOAK_REALM}} EFILING_DEMO_KEYCLOAK_URL=${{secrets.EFILING_DEMO_KEYCLOAK_URL}} docker-compose up -d efiling-frontend efiling-demo efiling-api redis sftp
+        # run: EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET=${{secrets.EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET}} KEYCLOAK_REALM=${{secrets.KEYCLOAK_REALM}} KEYCLOAK_URL=${{secrets.KEYCLOAK_URL}} EFILING_DEMO_KEYCLOAK_REALM=${{secrets.EFILING_DEMO_KEYCLOAK_REALM}} EFILING_DEMO_KEYCLOAK_URL=${{secrets.EFILING_DEMO_KEYCLOAK_URL}} docker-compose up -d efiling-frontend efiling-demo efiling-api redis sftp
+        run: |
+          sudo apt-get install -yqq nmap
+          EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET=${{secrets.EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET}} KEYCLOAK_REALM=${{secrets.KEYCLOAK_REALM}} KEYCLOAK_URL=${{secrets.KEYCLOAK_URL}} EFILING_DEMO_KEYCLOAK_REALM=${{secrets.EFILING_DEMO_KEYCLOAK_REALM}} EFILING_DEMO_KEYCLOAK_URL=${{secrets.EFILING_DEMO_KEYCLOAK_URL}} docker-compose up -d redis sftp
```
Comment (sdevalapurkar-bcgov) [inline .github/workflows/cucumber-tests.yml:25]:
```suggestion
          EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET=${{secrets.EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET}} KEYCLOAK_REALM=${{secrets.KEYCLOAK_REALM}} KEYCLOAK_URL=${{secrets.KEYCLOAK_URL}} EFILING_DEMO_KEYCLOAK_REALM=${{secrets.EFILING_DEMO_KEYCLOAK_REALM}} EFILING_DEMO_KEYCLOAK_URL=${{secrets.EFILING_DEMO_KEYCLOAK_URL}} docker-compose up -d redis sftp clamav
```

**Review by sdevalapurkar-bcgov — CHANGES_REQUESTED** [review]

File: .github/workflows/cucumber-tests.yml [hunk 2]
```diff
@@ -19,17 +19,27 @@ jobs:
...
+        run: |
+          sudo apt-get install -yqq nmap
+          EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET=${{secrets.EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET}} KEYCLOAK_REALM=${{secrets.KEYCLOAK_REALM}} KEYCLOAK_URL=${{secrets.KEYCLOAK_URL}} EFILING_DEMO_KEYCLOAK_REALM=${{secrets.EFILING_DEMO_KEYCLOAK_REALM}} EFILING_DEMO_KEYCLOAK_URL=${{secrets.EFILING_DEMO_KEYCLOAK_URL}} docker-compose up -d redis sftp clamav
+          EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET=${{secrets.EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET}} KEYCLOAK_REALM=${{secrets.KEYCLOAK_REALM}} KEYCLOAK_URL=${{secrets.KEYCLOAK_URL}} EFILING_DEMO_KEYCLOAK_REALM=${{secrets.EFILING_DEMO_KEYCLOAK_REALM}} EFILING_DEMO_KEYCLOAK_URL=${{secrets.EFILING_DEMO_KEYCLOAK_URL}} docker-compose up efiling-api &disown
+          sleep 30
+          docker ps
+          nmap -v 127.0.0.1
+          nmap -v 127.0.0.1 -p 8080
+          nmap -v 127.0.0.1 -p 6379
```
Comment (sdevalapurkar-bcgov) [inline .github/workflows/cucumber-tests.yml:31]:
```suggestion
          nmap -v 127.0.0.1 -p 6379
          nmap -v 127.0.0.1 -p 3310
```

Comment (sdevalapurkar-bcgov) [reply .github/workflows/cucumber-tests.yml:31]: adding clamav port

**Review by sdevalapurkar-bcgov — CHANGES_REQUESTED** [review]

File: .github/workflows/cucumber-tests.yml [hunk 3]
```diff
@@ -19,17 +19,28 @@ jobs:
 
       # Build Docker
       - name: build docker
-        run: KEYCLOAK_CREDENTIALS_SECRET=${{secrets.KEYCLOAK_CREDENTIALS_SECRET}} KEYCLOAK_REALM=${{secrets.KEYCLOAK_REALM}} KEYCLOAK_URL=${{secrets.KEYCLOAK_URL}} EFILING_DEMO_KEYCLOAK_REALM=${{secrets.EFILING_DEMO_KEYCLOAK_REALM}} EFILING_DEMO_KEYCLOAK_URL=${{secrets.EFILING_DEMO_KEYCLOAK_URL}} docker-compose up -d efiling-frontend efiling-demo efiling-api redis sftp
+        # run: EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET=${{secrets.EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET}} KEYCLOAK_REALM=${{secrets.KEYCLOAK_REALM}} KEYCLOAK_URL=${{secrets.KEYCLOAK_URL}} EFILING_DEMO_KEYCLOAK_REALM=${{secrets.EFILING_DEMO_KEYCLOAK_REALM}} EFILING_DEMO_KEYCLOAK_URL=${{secrets.EFILING_DEMO_KEYCLOAK_URL}} docker-compose up -d efiling-frontend efiling-demo efiling-api redis sftp
+        run: |
+          sudo apt-get install -yqq nmap
```
Comment (sdevalapurkar-bcgov) [inline .github/workflows/cucumber-tests.yml:24]:
```suggestion
          sudo apt-get install -yqq nmap
          sleep 10
```

**Review by sdevalapurkar-bcgov — CHANGES_REQUESTED** [review]

File: .github/workflows/cucumber-tests.yml [hunk 4]
```diff
@@ -19,17 +19,29 @@ jobs:
...
+          sleep 10
+          EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET=${{secrets.EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET}} KEYCLOAK_REALM=${{secrets.KEYCLOAK_REALM}} KEYCLOAK_URL=${{secrets.KEYCLOAK_URL}} EFILING_DEMO_KEYCLOAK_REALM=${{secrets.EFILING_DEMO_KEYCLOAK_REALM}} EFILING_DEMO_KEYCLOAK_URL=${{secrets.EFILING_DEMO_KEYCLOAK_URL}} docker-compose up -d redis sftp clamav
```
Comment (sdevalapurkar-bcgov) [inline .github/workflows/cucumber-tests.yml:26]:
```suggestion
          EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET=${{secrets.EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET}} KEYCLOAK_REALM=${{secrets.KEYCLOAK_REALM}} KEYCLOAK_URL=${{secrets.KEYCLOAK_URL}} EFILING_DEMO_KEYCLOAK_REALM=${{secrets.EFILING_DEMO_KEYCLOAK_REALM}} EFILING_DEMO_KEYCLOAK_URL=${{secrets.EFILING_DEMO_KEYCLOAK_URL}} docker-compose up -d redis sftp clamav
until (printf "PING\r\n";) | nc 127.0.0.1 6379 > /dev/null;
do
echo "Waiting for database..."
done
```

File: tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/requestbuilders/GenerateUrlRequestBuilders.java [hunk 5]
```diff
@@ -61,13 +66,16 @@ public Response requestWithSinglePdfDocument(String resourceValue, String accoun
 
         File pdfFile = new File(UPLOAD_FILE_PATH + fileNamePath);
 
+        ReadConfig readConfig = new ReadConfig();
+        String baseUri= readConfig.getBaseUri();
```
Comment (sdevalapurkar-bcgov) [inline tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/requestbuilders/GenerateUrlRequestBuilders.java:70]:
```suggestion
        String baseUri = readConfig.getBaseUri();
```

File: .github/workflows/cucumber-tests.yml [hunk 6]
```diff
@@ -19,17 +19,30 @@ jobs:
...
+          EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET=${{secrets.EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET}} KEYCLOAK_REALM=${{secrets.KEYCLOAK_REALM}} KEYCLOAK_URL=${{secrets.KEYCLOAK_URL}} EFILING_DEMO_KEYCLOAK_REALM=${{secrets.EFILING_DEMO_KEYCLOAK_REALM}} EFILING_DEMO_KEYCLOAK_URL=${{secrets.EFILING_DEMO_KEYCLOAK_URL}} CSO_DEBUG_ENABLED=false docker-compose build efiling-api
+          EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET=${{secrets.EFILING_DEMO_KEYCLOAK_CREDENTIALS_SECRET}} KEYCLOAK_REALM=${{secrets.KEYCLOAK_REALM}} KEYCLOAK_URL=${{secrets.KEYCLOAK_URL}} EFILING_DEMO_KEYCLOAK_REALM=${{secrets.EFILING_DEMO_KEYCLOAK_REALM}} EFILING_DEMO_KEYCLOAK_URL=${{secrets.EFILING_DEMO_KEYCLOAK_URL}} CSO_DEBUG_ENABLED=false docker-compose up efiling-api &disown
+          sleep 30
+          docker ps
+          nmap -v 127.0.0.1
```
Comment (sdevalapurkar-bcgov) [inline .github/workflows/cucumber-tests.yml:31]:
```suggestion
          curl http://localhost:8080
          curl http://localhost:8080/submission/documents
          nmap -v 127.0.0.1
```

File: tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/requestbuilders/GenerateUrlRequestBuilders.java (hunk 5 above)
Comment (karthikmsqa) [reply tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/requestbuilders/GenerateUrlRequestBuilders.java]: reverted this change. Not applicable.

File: tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/frontend/pages/LandingPage.java [hunk 7]
```diff
@@ -63,6 +63,7 @@ public void enterAccountGuid(String guid) {
     }
 
     public void clickEfilePackageButton() {
+
```
Comment (akroon3r) [inline tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/frontend/pages/LandingPage.java:66]:
```suggestion
```

File: tests/src/test/java/features/backendfeatures/1-generateUrlAndSubmissionForExistingGuid.feature [hunk 8]
```diff
@@ -15,7 +15,7 @@ Feature: When user uploads the documents, id is authenticated user details, navi
     Given "SUBMISSION" id is submitted with GET http request
     When status code is 200 and content type is verified
     Then verify clientAppName and csoBaseUrl values are returned and not empty
-    And verify success, error and cancel navigation urls are returned
+    #And verify success, error and cancel navigation urls are returned
```
Comment (akroon3r) [inline tests/src/test/java/features/backendfeatures/1-generateUrlAndSubmissionForExistingGuid.feature:18]:
```suggestion
```

Comment (akroon3r) [reply tests/src/test/java/features/backendfeatures/1-generateUrlAndSubmissionForExistingGuid.feature:18]: anything commented out should be removed if we're not using it

**Review by akroon3r — COMMENTED** [review]
couple of small things

File: tests/src/test/java/features/backendfeatures/1-generateUrlAndSubmissionForExistingGuid.feature (hunk 8 above)
Comment (karthikmsqa) [reply tests/src/test/java/features/backendfeatures/1-generateUrlAndSubmissionForExistingGuid.feature:18]: fixed.

**Review by akroon3r — APPROVED** [review]

File: .github/workflows/cucumber-tests.yml [hunk 9]
```diff
@@ -2,7 +2,7 @@ name: Cucumber Tests
 
 on:
   push:
-    branches: [master, qa-squash-backup]
+    branches: [master, qa-backend-updates]
```
Comment (sdevalapurkar-bcgov) [inline .github/workflows/cucumber-tests.yml:5]: this can be just 
```suggestion
    branches: [master]
```

**Review by sdevalapurkar-bcgov — CHANGES_REQUESTED** [review]

**Review by sdevalapurkar-bcgov — APPROVED** [review]
excellent

**Pull request merged**